sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactoryExpressionJsonConverter.cs

```diff
@@ -11,32 +15,157 @@ internal class DataFactoryExpressionJsonConverter : JsonConverter<object>
+            else if (IsGenericDataFactoryList(value.GetType()))
+            {
+                var methodInfo = typeof(DataFactoryExpressionJsonConverter)
+                    .GetMethod(
+                        "SerializeList",
+                        BindingFlags.Static | BindingFlags.NonPublic)!
+                    .MakeGenericMethod(value.GetType().GenericTypeArguments[0].GenericTypeArguments[0]);
+                methodInfo!.Invoke(null, new object[] { writer, value });
+            }
+            else
+            {
+                throw new InvalidOperationException($"Unable to convert {value.GetType().Name} into a DataFactoryExpression<T>");
+            }
+        }
+
+        private static bool IsGenericDataFactoryList(Type type)
+        {
+            return type.IsGenericType &&
+                   type.GetGenericTypeDefinition() == typeof(DataFactoryExpression<>) &&
+                   type.GenericTypeArguments[0].IsGenericType &&
+                   type.GenericTypeArguments[0].GetGenericTypeDefinition() == typeof(List<>) &&
+                   type.GenericTypeArguments[0].GenericTypeArguments[0].GetCustomAttributes().Any(a => a.GetType() == typeof(JsonConverterAttribute));
+        }
+
+        internal static void Serialize<T>(Utf8JsonWriter writer, DataFactoryExpression<T> expression)
+        {
+            if (expression.HasLiteral)
+            {
+                writer.WriteObjectValue(expression.LiteralInternal!);
+            }
+            else
+            {
+                SerializeExpression(writer, expression.Type!, expression.Expression!);
+            }
+        }
+
+        internal static void SerializeList<T>(Utf8JsonWriter writer, DataFactoryExpression<List<T>> expression)
+        {
+            if (expression.HasLiteral)
+            {
+                writer.WriteStartArray();
+                foreach (T? elem in expression.LiteralInternal!)
+                {
+                    JsonSerializer.Serialize(writer, elem);
+                }
+
+                writer.WriteEndArray();
+            }
+            else
+            {
+                SerializeExpression(writer, expression.Type!, expression.Expression!);
+            }
+        }
+
+        private static void SerializeExpression(Utf8JsonWriter writer, string type, string value)
+        {
+            writer.WriteStartObject();
+            writer.WritePropertyName("type");
+            writer.WriteStringValue(type);
+            writer.WritePropertyName("value");
+            writer.WriteStringValue(value);
+            writer.WriteEndObject();
+        }
+
+        internal static DataFactoryExpression<List<T>> DeserializeList<T>(JsonElement element)
+        {
+            if (element.ValueKind == JsonValueKind.Array)
+            {
+                var list = new List<T>();
+                foreach (var item in element.EnumerateArray())
+                {
+                    list.Add(JsonSerializer.Deserialize<T>(item.ToString()!)!);
```

Comment:
Are the `!`'s safe?  Is it possible an item.ToString() is `null`?  Is the second `!` even need since the api doesn't allow for nullable reference?

Reply:
I don't think ToString can be null in this case since we would be dealing with JSON objects in the array.

Reply:
It seems it can be null, so I've updated to use a `List<T?>`